geonode/resource/manager.py

```diff
@@ -648,84 +628,68 @@ def assignable_perm_condition(perm, resource_type):
                         else:
                             _permissions = copy.deepcopy(permissions)
 
-                        # default permissions for resource owner
-                        _perm_spec = set_owner_permissions(_resource, members=get_obj_group_managers(_owner))
+                        # Fixup Advanced Workflow permissions
+                        _perm_spec = AdvancedSecurityWorkflowManager.get_permissions(_resource.uuid, instance=_resource, permissions=_permissions)
```

Comment:
with a couple of changes, this part (the definition of the `_perm_spec` dictionary) can be port before the `if/else` condition. In this way in case of future changes is not needed to modify this part, can be done only once.

Reply:
done